scalalib/src/mill/scalalib/PublishModule.scala

```diff
@@ -252,8 +256,12 @@ trait PublishModule extends JavaModule { outer =>
 
 object PublishModule extends ExternalModule {
   val defaultGpgArgs: Seq[String] = Seq("--batch", "--yes", "-a", "-b")
+  val defaultStringGpgArgs: String = defaultGpgArgs.mkString(",")
 
-  case class PublishData(meta: Artifact, payload: Seq[(PathRef, String)])
+  case class PublishData(meta: Artifact, payload: Seq[(PathRef, String)]) {
+    def toDataWithConcretePath: (Seq[(Path, String)], Artifact) =
+      (payload.map { case (p, f) => (p.path, f) }, meta)
+  }
```

Comment:
I'd like some scaladoc here and maybe, you can flip the tuple so that `Artifact` comes first, so it's more clear it's just the unpacked `Path` version of the initial case class values.

Reply:
Unfortunately, this signature ```(Seq[(Path, String)], Artifact)``` is the one required by all of the publishAll signatures and refactoring it is starting to get out-of-scope for this PR. I agree that flipping it feels a little random here, but that isn't something that I introduced. This is just a helper function that centralizes what was already being done. I've added a scaladoc and made it `private[mill]`so that it can be refactored later.